File: AtCoder/ABC182/E.cs
using System;
using System.Linq;

class Program
{
  public static void Main()
  {
    var HWNM = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
    int H = HWNM[0], W = HWNM[1], N = HWNM[2], M = HWNM[3];
    var B = new int[H, W];
    var L = new bool[H, W];

    for (int i = 0; i < N; ++i)
    {
      var hw = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
      int h = hw[0] - 1, w = hw[1] - 1;
      B[h, w] = 1;
    }
    for (int i = 0; i < M; ++i)
    {
      var hw = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
      int h = hw[0] - 1, w = hw[1] - 1;
      B[h, w] = 2;
    }

    int ans = 0;
    bool on;
    for (int h = 0; h < H; ++h)
    {
      on = false;
      for (int w = 0; w < W; ++w)
      {
        int b = B[h, w];
        if (b == 1) on = true;
        if (b == 2) on = false;
        if (on && !L[h, w]) { L[h, w] = true; ++ans; }
      }

      on = false;
      for (int w = W - 1; w >= 0; --w)
      {
        int b = B[h, w];
        if (b == 1) on = true;
        if (b == 2) on = false;
        if (on && !L[h, w]) { L[h, w] = true; ++ans; }
      }
    }

    for (int w = 0; w < W; ++w)
    {
      on = false;
      for (int h = 0; h < H; ++h)
      {
        int b = B[h, w];
        if (b == 1) on = true;
        if (b == 2) on = false;
        if (on && !L[h, w]) { L[h, w] = true; ++ans; }
      }

      on = false;
      for (int h = H - 1; h >= 0; --h)
      {
        int b = B[h, w];
        if (b == 1) on = true;
        if (b == 2) on = false;
        if (on && !L[h, w]) { L[h, w] = true; ++ans; }
      }
    }

    Console.WriteLine(ans);
  }

}

File: AtCoder/ABC182/E.py
H, W, N, M = map(int, input().split())
B  = [[0]*W for _ in range(H)]
Bt = [[0]*H for _ in range(W)]

for _ in range(N):
  h, w = map(int, input().split())
  B[h-1][w-1] = Bt[w-1][h-1] = 1

for _ in range(M):
  h, w = map(int, input().split())
  B[h-1][w-1] = Bt[w-1][h-1] = 2

L = [[False]*W for _ in range(H)]

ans = 0

for h in range(H):
  Brow = B[h]
  on = False
  for w in range(W):
    b = Brow[w]
    if b == 1: on = True
    if b == 2: on = False
    if on and not L[h][w]:
      ans += 1
      L[h][w] = True
  
  on = False
  for w in range(W-1, -1, -1):
    b = Brow[w]
    if b == 1: on = True
    if b == 2: on = False
    if on and not L[h][w]:
      ans += 1
      L[h][w] = True

for w in range(W):
  Bcol = Bt[w]
  on = False
  for h in range(H):
    b = Bcol[h]
    if b == 1: on = True
    if b == 2: on = False
    if on and not L[h][w]:
      ans += 1
      L[h][w] = True
  
  on = False
  for h in range(H-1, -1, -1):
    b = Bcol[h]
    if b == 1: on = True
    if b == 2: on = False
    if on and not L[h][w]:
      ans += 1

print(ans)
